src/main/protobuf/spawn.proto

```diff
@@ -250,6 +258,40 @@ message ExecLogEntry {
     string target_path = 2;
   }
 
+  message RunfilesTree {
```

Comment:
@tjgq I haven't fixed all tests and would also add more, but could you already take a quick look at the general idea, in particular the proto representation?

Reply:
The way you're going with the proto representation SGTM; I might have some terminology nits to pick later when I do a full review, but feel free to proceed.

Reply:
Existing tests should be fixed now, I also added more documentation. 

Since some tests may need to assert on the exact proto structure, could we do a first round of review before I add them? Happy to resolve nits. :-)

Reply:
Added one more comment to explain the precedence in case of path collisions. Please doublecheck it, this part is pretty subtle.